[Look&Feel] Refactor to use semantic headers for page, modal & flyout

zhongnansu opened the pull request:
### Description
 Refactor to use semantic headers for page, modal & flyout

<!-- Describe what this change achieves-->
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/840aea11-c555-4ab2-8e0e-9b5c5250fcdd)



### Issues Resolved

<!-- List any issues this PR will resolve. Prefix the issue with the keyword closes, fixes, fix -->
<!-- Example: closes #1234 or fixes <Issue_URL> -->

## Screenshot

<!-- Attach any relevant screenshots. Any change to the UI requires an attached screenshot in the PR Description -->

## Testing the changes

<!--
  Please provide detailed steps for validating your changes. This could involve specific commands to run,
  pages to visit, scenarios to try or any other information that would help reviewers verify
  the functionality of your change
-->

## Changelog
<!--
Add a short but concise sentence about the impact of this pull request. Prefix an entry with the type of change they correspond to: breaking, chore, deprecate, doc, feat, fix, infra, refactor, test.
- fix: Update the graph
- feat: Add a new feature

If this change does not need to added to the changelog, just add a single `skip` line e.g.
- skip

Descriptions following the prefixes must be 100 characters long or less
-->
- refactor: [Look&Feel]  Refactor to use semantic headers for page, modal & flyout

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

virajsanghvi commented on src/plugins/saved_objects_management/public/management_section/object_view/components/header.tsx:
```diff
@@ -69,13 +70,16 @@ export const Header = ({
-              />
-            </h1>
+            // </EuiTitle>
+            <EuiTitle size="l">
+              <h1>
```
Do you have a screenshot of this one? This seems fine, but not sure

EuiText size='s'

zhongnansu replied:
same as above, will replace with euitext

virajsanghvi commented on src/plugins/saved_objects_management/public/management_section/objects_table/components/header.tsx:
```diff
@@ -54,7 +54,7 @@ export const Header = ({
     <EuiFlexGroup justifyContent="spaceBetween" alignItems="baseline">
       <EuiFlexItem grow={false}>
-        <EuiTitle>
+        <EuiTitle size="l">
```
EuiText size='s'

virajsanghvi commented on src/plugins/vis_default_editor/public/components/sidebar/sidebar_title.tsx:
```diff
@@ -100,7 +100,7 @@ export function LinkedSearch({ savedSearch, eventEmitter }: LinkedSearchProps) {
 
       <EuiFlexItem grow={false} className="eui-textTruncate">
-        <EuiTitle size="xs" className="eui-textTruncate">
+        <EuiTitle size="m" className="eui-textTruncate">
```
Let's not change these xs ones - if they're using that, they probably intentionally need to be very small.

virajsanghvi commented on src/plugins/vis_type_markdown/public/markdown_options.tsx:
```diff
@@ -56,7 +56,7 @@ function MarkdownOptions({ stateParams, setValue }: VisOptionsProps<MarkdownVisP
           <EuiFlexGroup gutterSize="none" justifyContent="spaceBetween" alignItems="baseline">
             <EuiFlexItem grow={false}>
-              <EuiTitle size="xs">
+              <EuiTitle size="m">
```
Same as above, let's not change

virajsanghvi commented on src/plugins/visualizations/public/wizard/type_selection/type_selection.tsx:
```diff
@@ -157,7 +157,7 @@ class TypeSelection extends React.Component<TypeSelectionProps, TypeSelectionSta
               ) : (
                 <React.Fragment>
-                  <EuiTitle size="s">
+                  <EuiTitle size="m">
```
This isn't really a header, let's not change it

virajsanghvi commented on src/plugins/visualizations/public/wizard/type_selection/vis_help_text.tsx:
```diff
@@ -42,7 +42,7 @@ interface VisHelpTextProps {
   return (
     <React.Fragment>
-      <EuiTitle size="s">
+      <EuiTitle size="m">
```
Is this in a visualization? If so, let's not change - given its intentionally small, not sure if we should change either way. If you can point of screenshot, would help me give feedback

zhongnansu replied:
this is a header in the select vis type modal, but it's not the major header. I agree we can leave it as small.
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/0ff39354-3c20-47e4-aeff-3e6ff98de730)

virajsanghvi commented on src/plugins/visualize/public/application/utils/get_table_columns.tsx:
```diff
@@ -167,12 +175,14 @@ export const getNoItemsMessage = (createItem: () => void) => (
+        <h1 id="visualizeListingHeading">
+          <FormattedMessage
+            id="visualize.listing.createNew.title"
+            defaultMessage="Create your first visualization"
```
Let's leave this as is because its in the visualization

zhongnansu replied:
this is the page content header for create visualization. Given the requirement, I think we should wrap it with `EuiText `
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/e9125235-f6f5-4341-8b7c-a77c1672097c)

virajsanghvi reviewed (commented):
Clarifications on guidance after going through these: 
- Let's only make changes to page content headers or modal/flyout headers - other h1/h2 tags, let's ignore as they may not be being used correctly or are there for accessibility purposes.
- Let's swap out EuiTitle for EuiText size='s' (or wrap when not present)
- If its using a small or xs EuiTitle, let's probably not change as they're intentionally small.

Also, I think you will want to search for all modals/flyouts to adjust their headers. I don't think they all use header tags today so may not be coming up in your search.

zhongnansu commented on src/plugins/inspector/public/ui/inspector_panel.tsx:
```diff
@@ -122,8 +122,8 @@ export class InspectorPanel extends Component<InspectorPanelProps, InspectorPane
             <EuiFlexItem grow={true}>
-              <EuiTitle size="s">
-                <h1>{title}</h1>
+              <EuiTitle size="m">
+                <h2>{title}</h2>
```
will do

zhongnansu commented on src/plugins/saved_objects_management/public/management_section/object_view/components/header.tsx:
```diff
@@ -61,6 +61,7 @@ export const Header = ({
       <EuiPageContentHeaderSection>
         <EuiTitle>
           {canEdit ? (
+            // <EuiTitle size="l">
```
by mistake, for this I'll replace with EuiText small since its a page content header h1
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/e704bfff-d271-4e50-898a-e9720b9dc25f)

zhongnansu commented:
> Clarifications on guidance after going through these:
> 
> * Let's only make changes to page content headers or modal/flyout headers - other h1/h2 tags, let's ignore as they may not be being used correctly or are there for accessibility purposes.
> * Let's swap out EuiTitle for EuiText size='s' (or wrap when not present)
> * If its using a small or xs EuiTitle, let's probably not change as they're intentionally small.
> 
> Also, I think you will want to search for all modals/flyouts to adjust their headers. I don't think they all use header tags today so may not be coming up in your search.

Updated the PR to use EuiText accordingly and refactored missing modals/flyouts to use semantic headers.

virajsanghvi commented:
Given this isn't a flyout but a pane, let's not make this change: 
<img width="815" alt="image" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/1366272/24fc0d25-6234-42ee-a6d9-1e58f3b4969b">

zhongnansu commented:
> Given this isn't a flyout but a pane, let's not make this change: <img alt="image" width="815" src="https://private-user-images.githubusercontent.com/1366272/347173540-24fc0d25-6234-42ee-a6d9-1e58f3b4969b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1Njk3OTAsIm5iZiI6MTcyMDU2OTQ5MCwicGF0aCI6Ii8xMzY2MjcyLzM0NzE3MzU0MC0yNGZjMGQyNS02MjM0LTQyZWUtYTZkOS0xZTU4ZjNiNDk2OWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDlUMjM1ODEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODBkNzYxOGNiODBhZjAxNjkzZjllOTJjNDYyOTcyMDQ1MjFkYzE4MmZmYzg0MjFlMThiMjUyMDk0YjJiMGFlMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.U_4FUkyx9A16YH-32bbQnDzAH9CqVhSVpzhrYCDSeKs">

@virajsanghvi This is actually being reverted and will remain "xs". I will correct the screenshot
